### PR TITLE
Fix a few issues with Plotter.show

### LINKED ISSUE #1621: `auto_close=False` not functioning
**Describe the bug, what's wrong, and what you expect:**

I have a number of existing scripts that worked on a Windows machine and an older version of pyVista (I do not know which version as that computer had hard drive issues, and was cleaned).

The problem seems to be that the initial call to `p.show(auto_close=False)` still closes the plotter, and subsequent calls to `p.show(screenshot='image.png')` return an error. 

-----

**To Reproduce**

This behavior is reproduced with the code included in the PyVista `Plotter()` documentation:

```py
import pyvista as pv
pl = pv.Plotter()
_ = pl.add_mesh(pv.Cube())
pl.show(auto_close=False)  
pl.show(screenshot='my_image.png')  
```
which results in the following error after closing the interactive window (by pressing the `q` button):

```
python test.py 
/home/mcgoldba/miniconda3/envs/xcfdv/lib/python3.9/site-packages/pyvista-0.31.3-py3.9.egg/pyvista/plotting/plotting.py:4070: UserWarning: `auto_close` ignored: by clicking the exit button, you have destroyed the render window and we have to close it out.
  warnings.warn("`auto_close` ignored: by clicking the exit button, "
Traceback (most recent call last):
  File "/home/mcgoldba/OpenFOAM/mgoldbach-7/run/202000045/sim04/test.py", line 7, in <module>
    pl.show(screenshot='test_img.png')
  File "/home/mcgoldba/miniconda3/envs/xcfdv/lib/python3.9/site-packages/pyvista-0.31.3-py3.9.egg/pyvista/plotting/plotting.py", line 3988, in show
    raise RuntimeError("This plotter has been closed and cannot be shown.")
RuntimeError: This plotter has been closed and cannot be shown.
```

**System Information:**

```py
# Get system info
import pyvista as pv
print(pv.Report())
```

```txt
--------------------------------------------------------------------------------
  Date: Mon Aug 30 11:26:55 2021 EDT

                OS : Linux
            CPU(s) : 24
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : Quadro M2000/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 460.84

  Python 3.9.5 (default, Jun  4 2021, 12:28:51)  [GCC 7.5.0]

           pyvista : 0.31.3
               vtk : 9.0.20210710
             numpy : 1.21.0
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.7
            meshio : 4.4.6
        matplotlib : 3.4.2
          colorcet : 1.0.0
--------------------------------------------------------------------------------
```

### LINKED ISSUE #1635: Screenshot silently fails with panel & ipygany backends
This screenshot MWE is [from the docs](https://dev.pyvista.org/examples/02-plot/screenshot.html#saving-screenshots). 
```py
import pyvista as pv
from pyvista import examples
import matplotlib.pyplot as plt
filename=examples.planefile
mesh=pv.read(filename)
plotter=pv.Plotter(off_screen=True)
plotter.add_mesh(mesh, color="orange")
plotter.show(screenshot='airplane.png')
plt.imshow(plotter.image)
plt.show()
```
This works depending on what is the default jupyter backend. With `pv.set_jupyter_backend('panel')` (or `show(...,jupyter_backend='panel')`) or `ipygany`, there will be no screenshot saved, and no error message from `show`.

Tthere will only be error later when getting `plotter.image`: "This plotter has not yet been setup and rendered with `show()`. Consider setting `off_screen=True` for off screen rendering."

I expect `show(...)` to report an error rather than failing silently, or (even better) select suitable backend by itself.

**To Reproduce**

Run the above with `plotter.show(screenshot='airplane.png',jupyter_backed='panel')` or call `pv.set_jupyter_backend('panel')` before `plotter.show`. The correct behavior can be obtained by writing `pv.show(...,jupyter_backend='none')`.

-----

**System Information:**
 Please run the following code wherever you are experiencing the bug and paste the output below. This report helps us track down bugs and it is critical to addressing your bug:

```py
# Get system info
import pyvista as pv
print(pv.Report())
```

```txt
--------------------------------------------------------------------------------
  Date: Sat Sep 04 09:28:28 2021 CEST

                OS : Linux
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
               RAM : 30.4 GiB
       Environment : Jupyter
        GPU Vendor : X.Org
      GPU Renderer : AMD Ryzen Embedded V1605B with Radeon Vega Gfx (RAVEN, DRM 3.40.0, 5.11.0-27-generic, LLVM 9.0.1)
       GPU Version : 4.6 (Core Profile) Mesa 20.0.4

  Python 3.8.10 (default, Jun  2 2021, 10:49:15)  [GCC 9.4.0]

           pyvista : 0.31.3
               vtk : 9.0.3
             numpy : 1.17.4
           imageio : 2.9.0
           appdirs : 1.4.3
            scooby : 0.5.7
            meshio : 4.4.6
        matplotlib : 3.1.2
             PyQt5 : 5.14.1
           IPython : 7.24.1
          colorcet : 1.0.0
        ipyvtklink : 0.2.1
             scipy : 1.3.3
              tqdm : 4.61.0
--------------------------------------------------------------------------------
```

### LINKED ISSUE #1665: Plotter.show() requires camera position to be returned if user wants the last image.
**Describe the bug, what's wrong, and what you expect:**

When running `pv.Plotter().show()` the parameter `return_img` only works if `return_cpos` is also `True`. 

It looks from the code more a 'feature' than a bug, as after asking if `return_img` or `screenshot` is `True`, the code then asks if `return_cpos` is `True`:

https://github.com/pyvista/pyvista/blob/8b27bab0edef82d681d8fa0f6d40a7544ece319f/pyvista/plotting/plotting.py#L4909

Maybe I don't know enough about the code base for why this might be needed, but in either way its very unintuitive as the image returned is just `None`. 

I would recommend either:

```py
if return_img or screenshot is True:
    if return_cpos:
        return self.camera_position, self.last_image
    else:
        return self.last_image
```

If this is just a bug. Or if it's a feature for some reason:

```py
if return_img or screenshot is True:
    if not return_cpos:
        # some user warning here they will get None with return_cpos=True
    else:
        return self.camera_position, self.last_image
```

-----

**To Reproduce**

Please include a code snippet to reproduce the bug in the block below:

```py

plot = pv.Plotter(window_size=(512, 512), off_screen=True)
plot.background_color=[1.,1.,1.]

plot.add_points(np.random.rand(100, 3), color=[0.9, 0.9, 0.9], render_points_as_spheres=True, point_size=2)

plot.view_xy()
img = plot.show(return_img=True)

```

### LINKED ISSUE #1877: `plotter.show(screenshot=...)` will not work if `notebook=True`
**Describe the bug, what's wrong, and what you expect:**

`plotter.show(screenshot=...)` will not work if `notebook=True`

-----

**To Reproduce**

Please include a code snippet to reproduce the bug in the block below:

```py
import pyvista as pv
import os

mesh = pv.Sphere()
plotter = pv.Plotter() # in a notebook, so `notebook` defaults to True
plotter.add_mesh(mesh)
plotter.show(screenshot='test.png')
assert os.path.exists('test.png')
```
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
/var/folders/2x/4df66vz13pv4hj_ysv23tyyc0000gn/T/ipykernel_48191/1720419012.py in <module>
      6 plotter.add_mesh(mesh)
      7 plotter.show(screenshot='test.png')
----> 8 assert os.path.exists('test.png')

AssertionError: 
```

### LINKED ISSUE #2953: Depth mapping not working on jupyterlab3
### Describe the bug, what's wrong, and what you expected.

I have tried running the following [example](https://docs.pyvista.org/examples/02-plot/image_depth.html) and I get the following error. While this example does not include the `off_screen = True` parameter I included it within `show()` but still get an error

### Steps to reproduce the bug.

```python 
 import pyvista as pv
from pyvista import examples

# Load an interesting example of geometry
mesh = examples.load_random_hills()

# Establish geometry within a pv.Plotter()
p = pv.Plotter()
p.add_mesh(mesh, color=True)
p.store_image = True  # permit image caching after plotter is closed
p.show()

zval = p.get_image_depth() 
 ```

### System Information

```shell
--------------------------------------------------------------------------`
AttributeError                            Traceback (most recent call last)
Input In [13], in <cell line: 1>()
----> 1 zval = p.get_image_depth()

File C:\Python\Miniconda\envs\test_3d\lib\site-packages\pyvista\plotting\plotting.py:3488, in BasePlotter.get_image_depth(self, fill_value, reset_camera_clipping_range)
   3485         zval[self._image_depth_null] = fill_value
   3486     return zval
-> 3488 self._check_rendered()
   3489 self._check_has_ren_win()
   3491 # Ensure points in view are within clipping range of renderer?

File C:\Python\Miniconda\envs\test_3d\lib\site-packages\pyvista\plotting\plotting.py:1241, in BasePlotter._check_rendered(self)
   1239 """Check if the render window has been shown and raise an exception if not."""
   1240 if not self._rendered:
-> 1241     raise AttributeError(
   1242         '\nThis plotter has not yet been set up and rendered '
   1243         'with ``show()``.\n'
   1244         'Consider setting ``off_screen=True`` 
   1245         'for off screen rendering.\n'
   1246     )

AttributeError: 
This plotter has not yet been set up and rendered with ``show()`
Consider setting ``off_screen=True`` for off screen rendering. '
```

### Screenshots

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #3548: `show(screenshot=...)` regression
### Describe the bug, what's wrong, and what you expected.

In previous versions of PyVista, passing the `screenshot` keyword argument to `Plotter.show()` would take a screenshot on-close so that the user could navigate the scene and position the camera for their screenshot.

This is no longer the case and the screenshot is saved on-show (as the scene is initialized).

This is a regression and is counterintuitive.

### Steps to reproduce the bug.

```python
import pyvista as pv

mesh = pv.Cone()

p = pv.Plotter(notebook=0)
p.add_mesh(mesh)
p.show(screenshot='foo.png')
```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Thu Nov 03 19:33:31 2022 MDT

                OS : Darwin
            CPU(s) : 8
           Machine : arm64
      Architecture : 64bit
               RAM : 24.0 GiB
       Environment : Jupyter
       File system : apfs
        GPU Vendor : Apple
      GPU Renderer : Apple M2
       GPU Version : 4.1 Metal - 76.3

  Python 3.9.13 (main, Aug 25 2022, 18:24:45)  [Clang 12.0.0 ]

           pyvista : 0.38.dev0
               vtk : 9.2.2
             numpy : 1.23.1
           imageio : 2.22.1
            scooby : 0.5.12
             pooch : v1.6.0
        matplotlib : 3.6.1
           IPython : 8.5.0
          colorcet : 3.0.1
           cmocean : 2.0
        ipyvtklink : 0.2.3
             scipy : 1.9.2
              tqdm : 4.64.1
            meshio : 5.3.4
        jupyterlab : 3.4.8
         pythreejs : Version unknown
--------------------------------------------------------------------------------
```

### Screenshots

| Window on close | saved screenshot |
|---|---|
| Screen Shot 2022-11-03 at 7 34 00 PM | foo |

### PR DESCRIPTION
**Pull request by banesullivan**
Fix a few issues with `Plotter.show()`

Resolves #1635, Resolves #3548, Resolves #2953, Resolves #1877, Resolves #1665, Resolves #1621

Some cleanups to improve the return values of `Plotter.show()` so that if `return_*` is set, the requested item will always be returned.

Further, this removes the `store_image` property as the whole point of `last_image` was always to have an image after rendering. I probably need to deprecate this instead of outright removing... 

@akaszynski and @pyvista/developers, I would appreciate a thorough review on this one as changing up the `show()` method nearly always has unexpected consequences

### CODE REVIEW

**Comment by banesullivan** [comment]
There is a very odd image regression failure showing up here... hoping that #3891 resolves it

**Review by akaszynski — APPROVED** [review]
A few minor changes, otherwise LGTM.

Since we're maintaining backwards compatibility as much as possible, I agree with the changes here and appreciate removing `store_image`. It always felt like a hack.

File: pyvista/plotting/plotting.py [hunk 1]
```diff
@@ -6218,9 +6243,11 @@ def show(
             This can also be set globally with
             :func:`pyvista.set_jupyter_backend`.
 
-        return_viewer : bool, optional
-            Return the jupyterlab viewer, scene, or display object
-            when plotting with jupyter notebook.
+        return_viewer : bool, default: False
+            Return the jupyterlab viewer, scene, or display object when
+            plotting with Jupiter notebook. When ``False`` and within a Jupiter
```
Comment (banesullivan) [inline pyvista/plotting/plotting.py:6248]:
```suggestion
            plotting with Jupyter notebook. When ``False`` and within a Jupyter
```

File: pyvista/plotting/plotting.py [hunk 2]
```diff
@@ -827,6 +827,31 @@ def renderer(self):
...
+
+        .. deprecated:: 0.38.0
+           ``store_image`` is no longer used. Images are automatically cached
+           as needed.
+
+        """
+        from pyvista.core.errors import DeprecationError
+
+        raise DeprecationError(
+            '`store_image` has been depreciated as of 0.38.0 and is no longer used.'
```
Comment (banesullivan) [inline pyvista/plotting/plotting.py:842]:
```suggestion
            '`store_image` has been deprecated as of 0.38.0 and is no longer used.'
```

File: pyvista/plotting/plotting.py [hunk 3]
```diff
@@ -827,6 +827,31 @@ def renderer(self):
...
+            ' Images are automatically cached as needed.'
+        )
+
+    @store_image.setter
+    def store_image(self, value):
+        from pyvista.core.errors import DeprecationError
+
+        raise DeprecationError(
+            '`store_image` has been depreciated as of 0.38.0 and is no longer used.'
```
Comment (banesullivan) [inline pyvista/plotting/plotting.py:851]:
```suggestion
            '`store_image` has been deprecated as of 0.38.0 and is no longer used.'
```

**Comment by banesullivan** [comment]
Thanks! Merging after CI turns green and will sync #3897

**Pull request merged**